nginx-controller/controller/controller.go

```diff
@@ -1222,6 +1229,19 @@ func (lbc *LoadBalancerController) getMinionsForMaster(master *nginx.IngressEx)
 			glog.Errorf("Ingress Resource %v/%v with the 'nginx.org/mergible-ingress-type' annotation set to 'minion' must contain a Path", ings.Items[i].Namespace, ings.Items[i].Name)
 			continue
 		}
+
+		for _, path := range ings.Items[i].Spec.Rules[0].HTTP.Paths {
+			for _, minion := range minions {
```

Comment:
to check for map conflicts, it is better to use a map[string]*nginx.IngressEx, where the map key is a path, rather iterating over an array of minions.

Reply:
Using map[sting]*extensions.Ingress, where key is the path, and value is the minion Ingress.